File: core/src/main/java/org/github/krandom/generator/games/dice/DiceType.java
/*
 * Copyright (c) 2026 krandom contributors
 *
 * Licensed under the MIT License. See LICENSE in the project root for license information.
 */
package org.github.krandom.generator.games.dice;

/**
 * Standard polyhedral dice types used in tabletop games.
 */
public enum DiceType {
    D4(4),
    D6(6),
    D8(8),
    D10(10),
    D12(12),
    D20(20);

    private final int sides;

    DiceType(int sides) {
        this.sides = sides;
    }

    /**
     * Number of faces on this die.
     */
    public int sides() {
        return sides;
    }
}

File: src/main/kotlin/org/github/krandom/games/dice/enum/DiceType.kt
package org.github.krandom.games.dice.enum

import org.github.krandom.games.dice.utils.DiceTypeUtils

enum class DiceType(val nrFaces: Int, val defaultValues: List<String>) {
    D4(4, DiceTypeUtils.generateExpected(4)),
    D6(6, DiceTypeUtils.generateExpected(6)),
    D8(8, DiceTypeUtils.generateExpected(8)),
    D10(10, DiceTypeUtils.generateExpected(10)),
    D12(12, DiceTypeUtils.generateExpected(12)),
    D20(20, DiceTypeUtils.generateExpected(20));
}
